option_test.go

```diff
@@ -235,23 +235,74 @@ func TestWithAthenzURL(t *testing.T) {
 		{
 			name: "set success",
 			args: args{
-				t: "dummy",
+				athenzURL: "dummy",
 			},
 			checkFunc: func(opt Option) error {
-				prov := &authorizer{}
-				if err := opt(prov); err != nil {
+				authz := &authorizer{}
+				if err := opt(authz); err != nil {
 					return err
 				}
-				if prov.athenzURL != "dummy" {
+				if authz.athenzURL != "dummy" {
 					return fmt.Errorf("invalid param was set")
 				}
 				return nil
 			},
 		},
+		{
+			name: "remove http:// prefix",
+			args: args{
+				athenzURL: "http://dummy",
+			},
+			checkFunc: func(opt Option) error {
+				authz := &authorizer{}
+				if err := opt(authz); err != nil {
+					return err
+				}
+
+				if authz.athenzURL != "dummy" {
+					return fmt.Errorf("cannot set athenz url")
+				}
+				return nil
+			},
+		},
+		{
+			name: "remove https:// prefix",
+			args: args{
+				athenzURL: "https://dummy",
+			},
+			checkFunc: func(opt Option) error {
+				authz := &authorizer{}
+				if err := opt(authz); err != nil {
+					return err
+				}
+
+				if authz.athenzURL != "dummy" {
+					return fmt.Errorf("cannot set athenz url")
+				}
+				return nil
+			},
+		},
+		{
+			name: "do not remove other protocol",
```

Comment:
@WindzCUHK 

This test detects schemas other than `https://` and `http://`.
However, in the logic of following link was not implemented it.
https://github.com/yahoojapan/athenz-authorizer/pull/25/commits/b8824f24558262cd646737bcd9b54450d42a232f#diff-4cecda73b06e0c180acae5250e973afeR25

Reply:
I think it is intend.

@kevindiu
Do you why we have this test case?
I think we should remove any scheme in the URL or throw error if the scheme is not `https`.

Reply:
@kevindiu 
https://github.com/yahoojapan/athenz-authorizer/pull/25#discussion_r305650166

so, this is not intended?
will fix